### PR TITLE
Adds database interface

### PR DESCRIPTION
**Pull request by cppcooper**
This PR is for adding functions to interface with the database. This initially seeks to implement a `DBInterface` class with static methods to use as functions around the codebase without worrying about global variables, but if it turns out there is little to be gained by doing it this way it may be removed in favour of a global connection variable.

### CODE REVIEW

**Comment by cppcooper** [comment]
When done this PR should resolve #48

**Comment by cppcooper** [comment]
Changed the base branch to feat-store to simplify the diff. As now the PR has 1 file change.

**Comment by brianzhouzc** [comment]
Bumping this pr. We need database access for #88

**Review by brianzhouzc — APPROVED** [review]
Works perfectly on my end. Successfully connected to DBInterface. Seems like Github action doesn't like it and is throwing errors.

**Review by iwanlevin — DISMISSED** [review]

**Comment by brianzhouzc** [comment]
Github actions issue related to Postgres is fixed in #95. However, unit test for pqxx isn't passing. @iwanlevin can you look into this?

**Comment by brianzhouzc** [comment]
Issue fixed and checks passed

**Review by brianzhouzc — APPROVED** [review]

File: src/db-interface.cpp [hunk 1]
```diff
@@ -4,11 +4,11 @@
 
 pqxx::connection* con = nullptr;
 
-void DBInterface::connect(std::string password) {
+void DBInterface::connect(std::string host, std::string password) {
     if(!con) {
         char buffer[256];
         //todo: we should build the connection string based on the json config
-        sprintf(buffer, "postgresql://postgres:%s@postgres:5432/orthanc", password.c_str());
+        sprintf(buffer, "postgresql://postgres:%s@%s:5432/orthanc", password.c_str(), host.c_str());
```
Comment (cppcooper) [inline src/db-interface.cpp:11]: looks solid

**Review by iwanlevin — APPROVED** [review]

**Pull request merged**